webpack.config.js

```diff
@@ -0,0 +1,49 @@
+const path = require('path');
+const webpack = require('webpack');
+const NgAnnotatePlugin = require('ng-annotate-webpack-plugin');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
+
+const ENV = process.env.npm_lifecycle_event;
+const isProd = ENV === 'build-js';
+
+const config = {
+    devtool: isProd ? 'source-map' : 'eval-source-map',
+    entry: {
+        'app.bundle': path.join(__dirname, '/public/js/index'),
+        'vendor.bundle': path.join(__dirname, 'public/js/vendor/index')
+    },
+    output: {
+        path: path.join(__dirname, '/public/build/js'),
+        filename: '[name].js',
+        publicPath: '/build/js'
+    },
+    module: {
+        loaders: [
+            {
+                test: /\.js?$/,
+                include: path.join(__dirname, '/public/js'),
+                exclude: [
+                    path.join(__dirname, '/public/js/vendor')
+                ],
+                loader: 'babel-loader'
+            }
+        ]
+    },
+    plugins: [
+        new CleanWebpackPlugin([path.join(__dirname, '/public/build/js/*')], {
```

Comment:
After the loaders above run (babel is the only one currently, which transpiles the code to ES5), these plugins will run the following:
- remove existing JS files in the build folder
- pull out "common" dependencies into a separate file (vendor.bundle.js)
- fix an issue with angular 1's dependency injection when minification is used
- minify JS